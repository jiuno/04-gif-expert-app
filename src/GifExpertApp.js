import React, {useState} from 'react';
import { AddCategory } from './Components/AddCategory';
import { GifGrid } from './Components/GifGrid';


const GifExpertApp = ( {defaultCategories = []} ) =>  {

    const [Categories, setCategories] = useState(defaultCategories);
    
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory SetCategories={setCategories}/>
            <hr />

            <button onClick = { () => setCategories( cats => ( ['Pokemon', ...cats ]))}>Agregar Pokemon</button>

            <ol>
                {Categories.map( (category) => 
                <GifGrid 
                key={category}
                category={ category } />)}
            </ol>
        
        </>      
            )
    
};

export default GifExpertApp;