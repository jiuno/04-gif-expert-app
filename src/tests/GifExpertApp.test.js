import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import GifExpertApp from '../GifExpertApp';



describe('Pruebas para <GifExpertApp />', () => {
    
    test('Debe renderizar <GifExpertApp />', () => {
    
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe mostrar una lista de categorias', () => {
        
        const categories = ['Mia', 'Nina'];

        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        


    })
    
    
})



