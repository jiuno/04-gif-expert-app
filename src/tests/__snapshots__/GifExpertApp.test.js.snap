// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    SetCategories={[Function]}
  />
  <hr />
  <button
    onClick={[Function]}
  >
    Agregar Pokemon
  </button>
  <ol>
    <GifGrid
      category="Mia"
      key="Mia"
    />
    <GifGrid
      category="Nina"
      key="Nina"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas para <GifExpertApp /> Debe renderizar <GifExpertApp /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    SetCategories={[Function]}
  />
  <hr />
  <button
    onClick={[Function]}
  >
    Agregar Pokemon
  </button>
  <ol />
</Fragment>
`;
